features2d: build fails on Visual Studio 2013 and earlier

features2d: build fails on Visual Studio 2013 and earlier

##### System information (version)

- OpenCV => master (b97b650ab32da61c71e815282260a1ecb6828181 and after)
- Operating System / Platform => Windows 7 Pro 64 bit
- Compiler => Visual Studio 2012, 2013

##### Detailed description
- When building OpenCV master branch on Visual Studio 2013 and earlier, the compilation fails saying following error
```
 C:\work\opencv\modules\features2d\src\fast.cpp(448): error C2065: 'uint32_t' : undeclared identifier [C:\work\opencv\build\modules\features2d\opencv_features2d.vcxproj]
 C:\work\opencv\modules\features2d\src\fast.cpp(448): error C2146: syntax error : missing ';' before identifier 'uthreshold' [C:\work\opencv\build\modules\features2d\opencv_features2d.vcxproj]
 C:\work\opencv\modules\features2d\src\fast.cpp(448): error C2065: 'uthreshold' : undeclared identifier [C:\work\opencv\build\modules\features2d\opencv_features2d.vcxproj]
 C:\work\opencv\modules\features2d\src\fast.cpp(448): error C2146: syntax error : missing ';' before identifier 'threshold' [C:\work\opencv\build\modules\features2d\opencv_features2d.vcxproj]
 C:\work\opencv\modules\features2d\src\fast.cpp(453): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int [C:\work\opencv\build\modules\features2d\opencv_features2d.vcxproj]
 C:\work\opencv\modules\features2d\src\fast.cpp(453): error C2143: syntax error : missing ';' before '*' [C:\work\opencv\build\modules\features2d\opencv_features2d.vcxproj]
 C:\work\opencv\modules\features2d\src\fast.cpp(453): error C2065: 'pscore' : undeclared identifier [C:\work\opencv\build\modules\features2d\opencv_features2d.vcxproj]
 C:\work\opencv\modules\features2d\src\fast.cpp(462): error C2065: 'uint32_t' : undeclared identifier [C:\work\opencv\build\modules\features2d\opencv_features2d.vcxproj]
 C:\work\opencv\modules\features2d\src\fast.cpp(462): error C2146: syntax error : missing ';' before identifier 'score' [C:\work\opencv\build\modules\features2d\opencv_features2d.vcxproj]
 C:\work\opencv\modules\features2d\src\fast.cpp(462): error C2065: 'score' : undeclared identifier [C:\work\opencv\build\modules\features2d\opencv_features2d.vcxproj]
 C:\work\opencv\modules\features2d\src\fast.cpp(462): error C2065: 'pscore' : undeclared identifier [C:\work\opencv\build\modules\features2d\opencv_features2d.vcxproj]
 C:\work\opencv\modules\features2d\src\fast.cpp(463): error C2065: 'score' : undeclared identifier [C:\work\opencv\build\modules\features2d\opencv_features2d.vcxproj]
 C:\work\opencv\modules\features2d\src\fast.cpp(463): error C2065: 'uthreshold' : undeclared identifier [C:\work\opencv\build\modules\features2d\opencv_features2d.vcxproj]
 C:\work\opencv\modules\features2d\src\fast.cpp(466): error C2065: 'int32_t' : undeclared identifier [C:\work\opencv\build\modules\features2d\opencv_features2d.vcxproj]
 C:\work\opencv\modules\features2d\src\fast.cpp(466): error C2146: syntax error : missing ')' before identifier 'score' [C:\work\opencv\build\modules\features2d\opencv_features2d.vcxproj]
 C:\work\opencv\modules\features2d\src\fast.cpp(466): error C2059: syntax error : ')' [C:\work\opencv\build\modules\features2d\opencv_features2d.vcxproj]
```

- The cause of the error is that old Visual Studio doesn't accept [```int32_t```](https://github.com/opencv/opencv/blob/b97b650ab32da61c71e815282260a1ecb6828181/modules/features2d/src/fast.cpp#L466), [```uint32_t```](https://github.com/opencv/opencv/blob/b97b650ab32da61c71e815282260a1ecb6828181/modules/features2d/src/fast.cpp#L448) or [```uint8_t```](https://github.com/opencv/opencv/blob/b97b650ab32da61c71e815282260a1ecb6828181/modules/features2d/src/fast.cpp#L453)
- I'll write a patch to avoid this build error later

##### Steps to reproduce
- Build master on Visual Studio 2013 or earlier

--- a/modules/features2d/src/fast.cpp
+++ b/modules/features2d/src/fast.cpp
@@ -445,12 +445,12 @@ static inline int hal_FAST(cv::Mat& src, std::vector<KeyPoint>& keypoints, int t
 
     cv::KeyPoint kpt(0, 0, 7.f, -1, 0);
 
-    uint32_t uthreshold = (uint32_t) threshold;
+    unsigned uthreshold = (unsigned) threshold;
 
     int ofs = 3;
 
     int stride = (int)suppressedScores.step;
-    const uint8_t* pscore = suppressedScores.data;
+    const unsigned char* pscore = suppressedScores.data;
 
     keypoints.clear();
 
@@ -459,11 +459,11 @@ static inline int hal_FAST(cv::Mat& src, std::vector<KeyPoint>& keypoints, int t
         kpt.pt.y = (float)(y);
         for (int x = ofs; x + ofs < suppressedScores.cols; ++x)
         {
-            uint32_t score = pscore[y * stride + x];
+            unsigned score = pscore[y * stride + x];
             if (score > uthreshold)
             {
                 kpt.pt.x = (float)(x);
-                kpt.response = (nonmax_suppression != 0) ? (float)((int32_t)score - 1) : 0.f;
+                kpt.response = (nonmax_suppression != 0) ? (float)((int)score - 1) : 0.f;
                 keypoints.push_back(kpt);
             }
         }

